inventory_management_system_api/main.py

```diff
@@ -44,20 +44,23 @@ async def custom_general_exception_handler(_: Request, exc: Exception) -> JSONRe
 
 
 @app.exception_handler(RequestValidationError)
-async def custom_validation_exception_handler(request: Request, exc: RequestValidationError) -> JSONResponse:
+async def custom_validation_exception_handler(_: Request, exc: RequestValidationError) -> JSONResponse:
     """
     Custom exception handler for FastAPI to handle `RequestValidationError`.
 
     This method is used to handle validation errors that occur when processing incoming requests in FastAPI. When a
     `RequestValidationError` is raised during request parsing or validation, this handler will be triggered to log the
     error and call `request_validation_exception_handler` to return an appropriate response.
 
-    :param request: The incoming HTTP request that caused the validation error.
+    :param _: Unused
     :param exc: The exception object representing the validation error.
     :return: A JSON response with validation error details.
     """
-    logger.exception(exc)
-    return await request_validation_exception_handler(request, exc)
+    # Exclude input as it may contain sensitive information which we do not want to log or return back to user
+    detail = {"detail": jsonable_encoder(exc.errors(), exclude={"input"})}
+    # The traceback here contains the exception message which includes the input hence it is been excluded.
+    logger.exception(detail, exc_info=False)
+    return JSONResponse(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, content=detail)
```

Comment:
I added this in but I wanted to check if we actually need it? We are not working with any sensitive data at an endpoint level like we do with usernames and passwords in the Auth component. The only case I can think it would be useful is if a user accidentally adds some sensitive data in their request, a validation request is raised, and that sensitive data ends up in the logs.

Reply:
> if a user accidently pastes a password into the name field, its not going to complain and will log it anyway for the post requests and then it will be in the database as well

That's true. Okay I will just revert these changes then, thanks.